id-repository/credential-service/src/main/java/io/mosip/credentialstore/util/RestUtil.java

```diff
@@ -140,12 +141,11 @@
             try {
                 HttpEntity<Object> httpEntity = (HttpEntity<Object>) requestType;
                 HttpHeaders httpHeader = httpEntity.getHeaders();
-                Iterator<String> iterator = httpHeader.keySet().iterator();
-                while (iterator.hasNext()) {
-                    String key = iterator.next();
-                    if (!(headers.containsKey("Content-Type") && key == "Content-Type"))
-                        headers.add(key, httpHeader.get(key).get(0));
-                }
+				for (String key : httpHeader.keySet()) {
+					String contentType = "Content-Type";
+					if (!(headers.containsKey(contentType) && key.equals(contentType)))
+						headers.add(key, Objects.requireNonNull(httpHeader.get(key)).get(0));
```

Comment:
Objects.requireNonNull should not be used which will throw NPE. instead, if condition to be used.